### DISCUSSION TITLE
EAC and negative costs

### DISCUSSION THREAD

**Discussion by tsmbland** [EnergySystemsModellingLab/MUSE_OS#316] [General]
I've been looking into the EAC objective as it's relevant for [one of the tutorials](https://muse-os.readthedocs.io/en/documentation/user-guide/add-agent.html), and I've been getting some strange behaviour. Not sure if this counts as a bug per se, which is why I bring it up here rather than creating an issue, but I think it's worth flagging anyway.

Just taking a very simple example, here's what happens if I take the `default` model and switch the single agent objective from LCOE to EAC:

![lcoe](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/ef50650d-7400-4627-9eab-ffab31037f93)
![eac](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/efd5a621-a5fb-464e-965c-b0cccd42a5e7)

The black dashed line is the capacity that the system needs to satisfy heating demand in the peak timeslice. In the LCOE case, installed capacity nicely matches this line. However, in the EAC case the agent ends up massively overinvesting in heat technologies above what is needed to satisfy demand. The capacity seems to reach the limits imposed in technodata file (`MaxCapacityAddition`, `MaxCapacityGrowth`, `TotalCapacityLimit`). If you increase (e.g. double) these limits then you can get even more extreme behaviour:

![eac-capacity-limits-2x](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/ae18a39b-2e8a-4325-adba-3d7d1e6c1f78)

I think this may have something to do with the fact that EAC is negative for the heating technologies (gasbioler and heatpump):

![eac-costs](https://github.com/EnergySystemsModellingLab/MUSE_OS/assets/23723407/abbd8303-fdf0-44aa-abb5-8e5f6e511302)

This is because EAC subtracts an (upper-bound) estimate of revenue, which in this case is very high due to the high initial cost of `heat` (100 $MUS). You then end up with the situation where the best way to 'minimise costs' in the investment algorithm is to invest as much as possible in technolgies with negative EAC. Mathematically this makes sense, but logically it's a bit nonsensical to invest in all this extra capacity which will just be sitting idle.

Part of the problem may lie in the EAC calculation, which seems to assume that technologies will be running at full capacity, which can give an inflated estimate for the revenue. In any case, as long as the EAC is negative it seems like you're bound to get this kind of behaviour where the agent invests as much as it possibly can, which doesn't seem realistic to me.

Not sure whether this is something worth thinking more about, or just something that users need to be aware of when using the EAC objective (or any other objective that can be negative). Perhaps this is something MUSE2 already plans to address?

Anyway, if anyone has any thoughts about this I'd be very interested to hear.

**Comment by tsmbland** [comment]
@dalonsoa

**Comment by dalonsoa** [comment]
I think your diagnosis of the meaningless simulation results makes sense. Now, I don't know if a negative EAC also makes sense and is an scenario that might actually happen. I don't know, maybe with heavily subsidised technologies or something like that is more profitable to install as much as you can.

@ahawkes , is this something that can actually happen or is there an error in the calculation and the EAC should be limited to positive values or something entirely different?

**Comment by ahawkes** [comment]
Hi there, thanks for this. Negative EAC would seem rather odd! I think it could only happen if commodity prices (of consumed commodities) was negative. And I struggle to imagine a scenario where that would happen. Can we dig into this?
________________________________
